### PR TITLE
PEAR-1694: Customize Columns Menu dropdown refactor

### PR DESCRIPTION
**Pull request by ocshawn**
## Description
JIRA: [PEAR-1694](https://gdc-ctds.atlassian.net/browse/PEAR-1694) Item 2 a effects main table components column ordering drop down 

- use mantine component for dropdown menu 
- use mantine component for Switch label on menu items 
- cleanup aria labels

## Checklist

- [x] Added proper unit tests
- [x] Left proper TODO messages for any remaining tasks
- [x] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

File: packages/portal-proto/src/components/Table/ColumnOrdering.tsx [hunk 1]
```diff
@@ -248,9 +248,6 @@ function DraggableColumnItem<TData>({
             }
           }}
           size="xs"
-          aria-label={`toggle column visibility switch button for ${humanify({
-            term: column.id,
-          })} column`}
```
Comment (paribartandhakal) [inline packages/portal-proto/src/components/Table/ColumnOrdering.tsx:253]: @bpierce01 did we use this aria-label in automation?

File: packages/portal-proto/src/components/Table/ColumnOrdering.tsx [hunk 2]
```diff
@@ -81,16 +86,18 @@ function ColumnOrdering<TData>({
...
-        <span>
+        <Menu.Target>
           <ActionIcon
             variant="outline"
             size="lg"
-            onClick={() => setShowColumnMenu((o) => !o)}
-            aria-label={`${
-              showColumnMenu ? "close" : "show"
-            } column change menu button`}
+            aria-label="Customize Columns"
```
Comment (paribartandhakal) [inline packages/portal-proto/src/components/Table/ColumnOrdering.tsx:100]: Nitpick: I feel like the prior was more informative. I am fine with it though. @ocshawn @wteouchicago

**Review by paribartandhakal — APPROVED** [review]
I left a few comments but looks good otherwise.

File: packages/portal-proto/src/components/Table/ColumnOrdering.tsx (hunk 2 above)
Comment (ocshawn) [reply packages/portal-proto/src/components/Table/ColumnOrdering.tsx:100]: when using the menu component it automatically adds a aria-expanded property this makes the inclusion of any text that describe the state unneeded as its handled by that

**Pull request merged**